magellan-library/src/main/java/com/wealthfront/magellan/lifecycle/ShownFieldFromContext.kt

```diff
@@ -0,0 +1,22 @@
+package com.wealthfront.magellan.lifecycle
+
+import android.content.Context
+
+public class ShownFieldFromContext<Field>(
+  public val fieldSupplier: (Context) -> Field
+) : LifecycleAware {
+
+  public var field: Field? = null
+    protected set
+
+  override fun show(context: Context) {
+    field = fieldSupplier(context)
+  }
+
+  override fun hide(context: Context) {
+    field = null
+  }
+}
+
+public fun <Field> LifecycleOwner.shownFieldFromContext(fieldSupplier: (Context) -> Field): LifecycleChild<ShownFieldFromContext<Field>, Field?> =
+  attachFieldToLifecycle(ShownFieldFromContext(fieldSupplier), { it.field })
```

Comment:
How do you feel about `createAndAttachFieldToLifecycleWhenShown`? It's very long but it's descriptive. I'm not sure `shownFieldFromContext` really says what's going on here. (Not that `lifecycleWithContext` was any better.)